src/Icicle/Avalanche/Prim/Eval.hs

```diff
@@ -111,7 +111,7 @@ evalPrim p vs
       -> primError
 
 
-     PrimUpdate (PrimUpdateArrayPut t)
+     PrimArray (PrimArrayPutMutable t)
```

Comment:
that's a shame. ah! I wonder whether the easiest way might be to actually create a new Statement for mutating put, instead of having mutating put as a Exp Flat Prim.
if it were on the statement level like WriteAccumulator etc currently are, it might be easier to modify the evaluator so you can update the scalar/exp heap there..